### ISSUE TITLE
SMTP connection error & diagnostic page says no internet access - related?

### ISSUE THREAD

**Issue by AndyYangUK** [dani-garcia/bitwarden_rs#1390]
# Please submit all your third party help requests to the forum
Link: https://bitwardenrs.discourse.group/c/help

Hi,
On my Bitwarden-rs installation, I've had it working for a while and it's located at https://bw.mydomain.com. I did initially have SMTP working, but at some point I've lost the config.json file through some folder reshuffles.

I tried to setup the bitwarden-rs container again, put my old database back into it and that works fine. I can log in, bitwarden can download favicons, etc... etc... 

**Issues:**
1. When I try to setup the SMTP again, I get error 400 Bad Request. 

_SMTP Settings:_
- Enabled: True
- Host: [mxroute server]
- Enable Secure SMTP: True
- Port: 465
- From address: admin@mydomain.com
- From Name: bitwarden_rs
- Username: admin@mydomain.com
- Password: [password] _I've checked and tested elsewhere that I can connect to this account_
- SMTP Auth mechanism: Plain

_Error:_
Error sending SMTP test email.
SMTP Client error: Could not connect.

2. I checked the Diagnostics tab to see what could be the issue, and it says I have no Internet Access, but everything else seems to be fine. 

As bitwarden can download favicons when I log in, and the fact that I can access it via public internet, doesn't that already suggest it has internet access?

Any help with this be helpful.

Thanks

**Comment by BlackDex** [comment]
Hello @AndyYangUK,
You closed this issue, could you state why and if you have solved this issue for other people.

**Comment by AndyYangUK** [comment]
Hi @BlackDex 

I'm not entirely sure exactly what it was that fixed this particular issue as I was having a couple other non-Bitwarden issues I was trying to resolve as well, but it is now fixed. I can send test emails and the diagnostics page shows I have internet access.

Below are a few things I did, I tried to do them iteratively to figure out what the exact issue was but life got in the way and I lost track of what I was doing, so this is just a dump of actions I did that eventually lead to a fix - but not in any particular order:

- Restart Docker
- Rebuilt the container from the latest image.
- In the docker daemon.json file, I had an iptables flag set to false. I removed that and restarted docker daemon.
- Restarted server
- Added the iptables flag as false again in the daemon.json. 
- Adjusted my UFW rules: allow from 172.17.0.0/16 to anywhere & allow from 172.16.0.0/16 to anywhere 

Sorry I can't be more helpful with this one!
